tree/ntuple/v7/doc/specifications.md

```diff
@@ -677,6 +677,13 @@ The on-disk representation is similar to a `std::vector<T>` where `T` is the val
   - Child field of type `T`, which must by a type with RNTuple I/O support.
     The name of the child field is `_0`.
 
+### ROOT::Experimental::RNTupleCardinality
+
+A field whose type is `ROOT::Experimental::RNTupleCardinality` should always appear as a "projected" field whose single alias column is the index column of a collection (see Section "Alias columns").
```

Comment:
I wouldn't restrict ourselves too much. This field can may appear in more contexts and at can be, at some point, a writable field, too. What counts for the specification is that it is represented by an index column, which it presents/interprets as offset lengths (and vice versa: lengths to offsets, should it come to writing).

Reply:
https://root.cern.ch/doc/master/RField_8hxx_source.html#l00904

says it's only for reading